[CI] org.elasticsearch.client.documentation.ReindexDocumentationIT.testReindex

This failed on 7.x

```
REPRODUCE WITH: ./gradlew ':modules:reindex:integTestRunner' --tests "org.elasticsearch.client.documentation.ReindexDocumentationIT.testReindex" -Dtests.seed=4566ED9FD0A1308B -Dtests.security.manager=true -Dtests.locale=et -Dtests.timezone=Pacific/Galapagos -Dcompiler.java=14 -Druntime.java=8
```

```

org.elasticsearch.client.documentation.ReindexDocumentationIT > testReindex FAILED
    java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 368
    Expected: <0L>
         but: was <368L>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$39(InternalTestCluster.java:2500)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:918)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:891)
        at org.elasticsearch.test.InternalTestCluster.assertRequestsFinished(InternalTestCluster.java:2497)
        at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2475)
        at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:602)
        at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2218)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.lang.Thread.run(Thread.java:748)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=571, name=elasticsearch[node_sd4][generic][T#5], state=RUNNABLE, group=TGRP-ReindexDocumentationIT]

        Caused by:
        java.lang.AssertionError: Unexpected AlreadyClosedException
            at __randomizedtesting.SeedInfo.seed([4566ED9FD0A1308B]:0)
            at org.elasticsearch.index.engine.InternalEngine.failOnTragicEvent(InternalEngine.java:2087)
            at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:1637)
            at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:1599)
            at org.elasticsearch.index.shard.IndexShard.finalizeRecovery(IndexShard.java:1716)
            at org.elasticsearch.indices.recovery.RecoveryTarget.lambda$finalizeRecovery$1(RecoveryTarget.java:313)
            at org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:325)
            at org.elasticsearch.indices.recovery.RecoveryTarget.finalizeRecovery(RecoveryTarget.java:294)
            at org.elasticsearch.indices.recovery.PeerRecoveryTargetService$FinalizeRecoveryRequestHandler.messageReceived(PeerRecoveryTargetService.java:397)
            at org.elasticsearch.indices.recovery.PeerRecoveryTargetService$FinalizeRecoveryRequestHandler.messageReceived(PeerRecoveryTargetService.java:391)
            at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
            at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:272)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:695)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
            at java.lang.Thread.run(Thread.java:748)

            Caused by:
            org.apache.lucene.store.AlreadyClosedException: engine is closed
                at org.elasticsearch.index.shard.IndexShard.getEngine(IndexShard.java:2539)
                at org.elasticsearch.index.shard.IndexShard$RefreshPendingLocationListener.beforeRefresh(IndexShard.java:3291)
                at org.apache.lucene.search.ReferenceManager.notifyRefreshListenersBefore(ReferenceManager.java:279)
                at org.apache.lucene.search.ReferenceManager.doMaybeRefresh(ReferenceManager.java:175)
                at org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking(ReferenceManager.java:253)
                at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:1622)
                ... 14 more

```

https://gradle-enterprise.elastic.co/s/lqcyi2m4orywg/console-log/raw

--- a/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java
+++ b/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java
@@ -3222,12 +3222,17 @@ public class IndexShard extends AbstractIndexShardComponent implements IndicesCl
 
         @Override
         public void beforeRefresh() {
-            lastWriteLocation = getEngine().getTranslogLastWriteLocation();
+            try {
+                lastWriteLocation = getEngine().getTranslogLastWriteLocation();
+            } catch (AlreadyClosedException exc) {
+                // shard is closed - no location is fine
+                lastWriteLocation = null;
+            }
         }
 
         @Override
         public void afterRefresh(boolean didRefresh) {
-            if (didRefresh) {
+            if (didRefresh && lastWriteLocation != null) {
                 pendingRefreshLocation.updateAndGet(pendingLocation -> {
                     if (pendingLocation == null || pendingLocation.compareTo(lastWriteLocation) <= 0) {
                         return null;

--- a/x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
+++ b/x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
@@ -170,7 +170,6 @@ public class AsyncSearchActionIT extends AsyncSearchIntegTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/55792")
     public void testRestartAfterCompletion() throws Exception {
         final AsyncSearchResponse initial;
         try (SearchResponseIterator it =
@@ -208,7 +207,6 @@ public class AsyncSearchActionIT extends AsyncSearchIntegTestCase {
         ensureTaskRemoval(response.getId());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/55792")
     public void testCleanupOnFailure() throws Exception {
         final AsyncSearchResponse initial;
         try (SearchResponseIterator it =

